lib/open_api_spex/cast/error.ex

```diff
@@ -0,0 +1,358 @@
+defmodule OpenApiSpex.Cast.Error do
+  alias OpenApiSpex.TermType
+
+  @type all_of_error :: {:all_of, [String.t()]}
+  @type any_of_error :: {:any_of, [String.t()]}
+  @type exclusive_max_error :: {:exclusive_max, non_neg_integer(), non_neg_integer()}
+  @type exclusive_min_error :: {:exclusive_min, non_neg_integer(), non_neg_integer()}
+  @type invalid_enum_error :: {:invalid_enum}
+  @type invalid_format_error :: {:invalid_format, any()}
+  @type invalid_schema_error :: {:invalid_schema_type}
+  @type invalid_type_error :: {:invalid_type, String.t() | atom()}
+  @type max_items_error :: {:max_items, non_neg_integer(), non_neg_integer()}
+  @type max_length_error :: {:max_length, non_neg_integer()}
+  @type max_properties_error :: {:max_properties, non_neg_integer(), non_neg_integer()}
+  @type maximum_error :: {:maximum, integer(), integer()}
+  @type min_items_error :: {:min_items, non_neg_integer(), non_neg_integer()}
+  @type min_length_error :: {:min_length, non_neg_integer()}
+  @type minimum_error :: {:minimum, integer(), integer()}
+  @type missing_field_error :: {:missing_field, String.t() | atom()}
+  @type multiple_of_error :: {:multiple_of, non_neg_integer(), non_neg_integer()}
+  @type no_value_for_discriminator_error :: {:no_value_for_discriminator, String.t() | atom()}
+  @type invalid_discriminator_value_error :: {:invalid_discriminator_value, String.t() | atom()}
+  @type null_value_error :: {:null_value}
+  @type one_of_error :: {:one_of, [String.t()]}
+  @type unexpected_field_error :: {:unexpected_field, String.t() | atom()}
+  @type unique_items_error :: {:unique_items}
```

Comment:
Any reason to make it single atom tuple instead of making it just an atom?

Reply:
@hauleth I discussed this with @moxley and the idea was to remain consistent. If you would prefer the atom then we can make that adjustment. There's a couple others here that will need to be adjusted ass well.

Reply:
It's consistent with the other types above. All of them are a tuple with at least an atom as the first element.

Reply:
@hauleth just wanted to follow up on this. Is the explanation acceptable or would you prefer the change to an atom?